pkg/resource/deploy/deploytest/pluginhost.go

```diff
@@ -112,16 +138,24 @@ func (host *pluginHost) SignalCancellation() error {
 	return err
 }
 func (host *pluginHost) Close() error {
+	host.m.Lock()
+	defer host.m.Unlock()
+
+	host.closed = true
 	return nil
 }
 func (host *pluginHost) ServerAddr() string {
 	panic("Host RPC address not available")
 }
 func (host *pluginHost) Log(sev diag.Severity, urn resource.URN, msg string, streamID int32) {
-	host.sink.Logf(sev, diag.StreamMessage(urn, msg, streamID))
+	if !host.isClosed() {
```

Comment:
It seems to me that `host.isClosed()` and the write to the sink all need to be protected by a lock in order to be truly safe in the face of another goroutine closing us while logging, right? Otherwise someone else could close us right after we checked `!host.isClosed()` and we still crash.

Reply:
That's right. It didn't concern me too much when I wrote the code, as it's strictly for a test that avoids that scenario. It might be better to just jam a new sink in during `Close()`--at the very least, it would be more reliable.